### ISSUE TITLE
Animepahe server cannot be found

### ISSUE THREAD

**Issue by ayushjaipuriyar** [vn-ki/anime-downloader#412]
**Describe the bug**
Can't download from animepahe it just says

*ERROR: 503 Server Error: Service Temporarily Unavailable for url: https://animepahe.com/api?l=8&m=search&q=inuyasha*

**To reproduce**

anime dl inuyasha --provider animpahe -q 480p
and the error received is 

ERROR: 503 Server Error: Service Temporarily Unavailable for url: https://animepahe.com/api?l=8&m=search&q=inuyasha

this log level debug doesn't really work
<details>
<summary>LOG</summary>

```
On searching the url given in the error 

the browser shows 

{"total":1,"data":[{"id":1711,"slug":"inuyasha","title":"InuYasha (TV)","type":"TV","episodes":167,"status":"Finished Airing","season":"Fall","year":2000,"score":7.85,"poster":"https:\/\/i.animepahe.com\/posters\/9dc06bac320b0d778be645bff712f7d1.jpg","session":"d66bd27c-a074-9a98-14bf-0ad269def8fe","relevance":"38"}]}


```

**Comment by Blatzar** [comment]
Yeah animepahe recently got cloudflare, will probably be able to fix with selenium.

**Comment by Blatzar** [comment]
Looked into it, animepahe has cloudflare AND hcaptcha which will be promoted EVERY time you visit the site (wtf is this fort knox?). Cracking this is definitely possible, but it will take some time.

**Comment by ayushjaipuriyar** [comment]
Okay I get the fact that it updates every time I open up the site but for some reason as such on the browser site works fine but it shows error on animedl

**Comment by ayushjaipuriyar** [comment]
Also selenium doesnt fix the issue

**Comment by Blatzar** [comment]
> Also selenium doesnt fix the issue

I'm well aware, it's just a step towards a fix. Before testing I thought it would solve since I've never seen a site use both cloud-flare and hcaptcha.

>AND hcaptcha

Selenium can get past the cloud-flare, @AbdullahM0hamed tool to bypass hcaptcha just needs to get added correctly and the site will probably work.

**Comment by ayushjaipuriyar** [comment]
Ohh well let me try then 
Thanks for anime-downloader though

**Comment by LordShenron** [comment]
Waiting for the fix now ^_^ 
Luckily I had downloaded a lot of anime for now.

**Comment by ayushjaipuriyar** [comment]
I see no god up here .
having 2tb of anime and shows

**Comment by ayushjaipuriyar** [comment]
> Waiting for the fix now ^_^
> Luckily I had downloaded a lot of anime for now.

dont worry you cant be above me had 2tb of anime

**Comment by LordShenron** [comment]
> > Waiting for the fix now ^_^
> > Luckily I had downloaded a lot of anime for now.
> 
> dont worry you cant be above me had 2tb of anime

I mostly stream and and delete the anime after if i download after watching.. but still 2TB is too much xD

**Comment by ayushjaipuriyar** [comment]
> > > Waiting for the fix now ^_^
> > > Luckily I had downloaded a lot of anime for now.
> > 
> > 
> > dont worry you cant be above me had 2tb of anime
> 
> I mostly stream and and delete the anime after if i download after watching.. but still 2TB is too much xD

yeah well first i thought that i wont be able to fill up 1tb hdd but then you know after watching long series hehehe .

well do you know any other site that provides mini hd and supported by anime-dowloader

**Comment by LordShenron** [comment]
> > > > Waiting for the fix now ^_^
> > > > Luckily I had downloaded a lot of anime for now.
> > > 
> > > 
> > > dont worry you cant be above me had 2tb of anime
> > 
> > 
> > I mostly stream and and delete the anime after if i download after watching.. but still 2TB is too much xD
> 
> yeah well first i thought that i wont be able to fill up 1tb hdd but then you know after watching long series hehehe .
> 
> well do you know any other site that provides mini hd and supported by anime-dowloader

Whenever animepahe breaks i use animeout.xyz, its supported by anime downloader and the sizes are just a little bigger than animepahe, but in my opinion quality is better too.

**Comment by ayushjaipuriyar** [comment]
> > > > > Waiting for the fix now ^_^
> > > > > Luckily I had downloaded a lot of anime for now.
> > > > 
> > > > 
> > > > dont worry you cant be above me had 2tb of anime
> > > 
> > > 
> > > I mostly stream and and delete the anime after if i download after watching.. but still 2TB is too much xD
> > 
> > 
> > yeah well first i thought that i wont be able to fill up 1tb hdd but then you know after watching long series hehehe .
> > well do you know any other site that provides mini hd and supported by anime-dowloader
> 
> Whenever animepahe breaks i use animeout.xyz, its supported by anime downloader and the sizes are just a little bigger than animepahe, but in my opinion quality is better too.

but the speed 300kbps : (

**Comment by ngomile** [comment]
Animepahe seems to be functional again but the bypassing seems to be incorrect now like hcaptcha has changed something @AbdullahM0hamed 
![Capture](https://user-images.githubusercontent.com/39145128/85947743-d925af00-b94c-11ea-962e-57c3559a51e8.PNG)

**Comment by AbdullahM0hamed** [comment]
They changed a whole lot, added browser checks, rate limiting, the works. Still haven't succeeded with the new one.

**Comment by LordShenron** [comment]
Aren't they doing way too much for pirating ;_; Just yesterday I had to wait hours because i couldn't solved their hcaptcha

**Comment by ngomile** [comment]
Could be either of these things or perhaps how to acquire the challenges from https://hcaptcha.com/getcaptcha has been changed. But at least animepahe is accessible again but now the hcaptcha problem.

**Comment by AbdullahM0hamed** [comment]
Perhaps we should consider having ffmpeg as an optional dependendency and getting the m3u8 stream? Seems like a more plausible option to me.

**Comment by AbdullahM0hamed** [comment]
So this works: https://github.com/JimmyLaurent/hcaptcha-solver/blob/master/index.js, and it doesn't give any of the browser stuff. Haven't been able to replicate in python thus far, though

**Comment by ngomile** [comment]
I tried using it like so solveCaptcha('https://kwik.cx/f/v7WXWU6dq0fL').then(console.log).catch(console.error) but it isn't giving me anything useful. I think it might have stopped working.

**Comment by AbdullahM0hamed** [comment]
It still works, and on that example. After a while it outputs the pass token if you use it in the node console. That being said, I've managed to actually get the captcha, and will probably working on finishing it tomorrow (hopeful for success, not certain) - but it will take a bit longer than the previous version because of rate limiting.

**Comment by LordShenron** [comment]
> but the speed 300kbps : (

@ayushjaipuriyar I actually tried many sites past few days and animefreak seems to be working just as good as animepahe, its fast and encodes are smaller.

**Comment by ayushjaipuriyar** [comment]
> > but the speed 300kbps : (
> 
> @ayushjaipuriyar I actually tried many sites past few days and animefreak seems to be working just as good as animepahe, its fast and encodes are smaller.

still bigger i would say cause animepahe gives you 720p most of the time under 100mb-120mb freak is around 140-170mb
nevertheless better than nothing

**Comment by LordShenron** [comment]
> > > but the speed 300kbps : (
> > 
> > 
> > @ayushjaipuriyar I actually tried many sites past few days and animefreak seems to be working just as good as animepahe, its fast and encodes are smaller.
> 
> still bigger i would say cause animepahe gives you 720p most of the time under 100mb-120mb freak is around 140-170mb
> nevertheless better than nothing

yes some newer currently airing anime are heavy in size while some are very good like, i downloaded "yesterday wo utatte" and it was 1.1gig for 13 ep, I would say its almost on par with animepahe.

**Comment by AbdullahM0hamed** [comment]
So, I've been trying for a few days to get the hcaptcha token, and despite sending all of the correct data, I've unable to actually succeed at getting a token on any of my tries (maybe because of the fact that the requests fucks up the formatting of at least one thing, 100% of the time), so I've decided to give up.

**Comment by ngomile** [comment]
Using the hcaptcha-solver module I've found that just letting it get the token and passing it to a function I made, this one:
```js
async function generateFormData(url, token, headers = {}) {
    headers = { ...headers, Referer: url };
    let response = await request.get(url, { headers, ...OPTIONS });
    const { body } = response;
    const [, action] = body.match(/action="([^"]+)/);
    const { origin } = new URL(url);
    const bypassURL = `${origin}${action}`;

    const $ = cheerio.load(body);
    let form = {};
    $('form > input').each(function (ind, elem) {
        form[$(this).attr('name')] = $(this).attr('value');
    });

    Object.assign(form, {
        "id": $('strong').first().text(),
        "g-recaptcha-response": token,
        "h-captcha-response": token
    });

    return { bypassURL, form };
}
```
So generating the token if you are attempting to do it Python side might just be a matter of something just being slightly wrong but yeah tough to solve Python side. But I've found that on JavaScript it works like a charm.

**Comment by AbdullahM0hamed** [comment]
Yeah, that's been the most frustrating thing. I've also been using the javascript of hcaptcha-solver as a point of reference, and everything looks the same when I monitor the requests😭😭😭

**Comment by AbdullahM0hamed** [comment]
Maybe I'll try again, but I wouldn't count on me succeeding.

**Comment by LordShenron** [comment]
I just noticed that animepahe don't ask for human verification while accessing the website anymore. Just to be sure i tried it on multiple devices and it was not asking on any of them

**Comment by AbdullahM0hamed** [comment]
Kwik gives hcaptcha, thought, and pretty much all of animepahe's videos are hosted on it.

**Comment by LordShenron** [comment]
> Kwik gives hcaptcha, though, and pretty much all of animepahe's videos are hosted on it.

Isn't that already bypassed by you, I thought pahe was broken because of human verification while accessing the site.

**Comment by Blatzar** [comment]
If they haven't changed much since the last time I checked you can probably access an m3u8 file used by the pahe without captcha.

**Comment by LordShenron** [comment]
> If they haven't changed much since the last time I checked you can probably access an m3u8 file used by the pahe without captcha.

I think they were testing hcaptcha while opening site too, but it was obviously stupid coz they already have that on kwik so now they removed it. But strangely it still don't work.

I am pasting the logs here with hopes that maybe something is changed. 
https://del.dog/animepahe-log.txt

**Comment by nate-moo** [comment]
nope the same as before

**Comment by C0nvert** [comment]
> If they haven't changed much since the last time I checked you can probably access an m3u8 file used by the pahe without captcha.

I can't find animepahe anymore as a listed provider.
Is it sstill supported.
Can you maybe explain in detail how to get the m3u8 access ?

**Comment by LordShenron** [comment]
> > If they haven't changed much since the last time I checked you can probably access an m3u8 file used by the pahe without captcha.
> 
> I can't find animepahe anymore as a listed provider.
> Is it sstill supported.
> Can you maybe explain in detail how to get the m3u8 access ?

You can use WFD downloader to download from animepahe.

**Comment by C0nvert** [comment]
@LordShenron 
Would you be so kind to post a Link to it.
Google Search for "WFD Downloader" didn't give me any viable results.

**Comment by LordShenron** [comment]
> @LordShenron
> Would you be so kind to post a Link to it.
> Google Search for "WFD Downloader" didn't give me any viable results.

https://www.wfdownloader.xyz/
here

**Comment by uali6981** [comment]
@C0nvert If you want m3u8 file link then use [this](https://github.com/KevCui/animepahe-dl)

**Comment by IshigamiYuu** [comment]
> @C0nvert If you want m3u8 file link then use [this](https://github.com/KevCui/animepahe-dl)

how to install this?

**Comment by uali6981** [comment]
> how to install this?

install all Dependency manually and then run
